File: backend/src/main/java/com/documind/entity/Document.java
package com.documind.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "documents")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Document {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private Profile user;

    @Column(name = "file_name", nullable = false)
    private String fileName;

    @Column(name = "storage_path", nullable = false, length = 500)
    private String storagePath;

    @Column(name = "file_size", nullable = false)
    private long fileSize;

    @Column(name = "page_count")
    private Integer pageCount;

    @Column(name = "chunk_count")
    private Integer chunkCount;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 50)
    @Builder.Default
    private DocumentStatus status = DocumentStatus.UPLOADING;

    @Column(name = "processing_progress", nullable = false)
    @Builder.Default
    private int processingProgress = 0;

    @Column(name = "error_message", columnDefinition = "TEXT")
    private String errorMessage;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;
}

File: src/types/document.ts
export type DocumentStatus =
  | "UPLOADING"
  | "EXTRACTING"
  | "CHUNKING"
  | "EMBEDDING"
  | "INDEXING"
  | "PROCESSING"
  | "COMPLETED"
  | "FAILED";

export interface DocumentItem {
  id: string;
  fileName: string;
  storagePath?: string;
  fileSize: number;
  pageCount?: number | null;
  chunkCount?: number | null;
  status: DocumentStatus;
  processingProgress?: number | null;
  errorMessage?: string | null;
  createdAt: string;
  updatedAt?: string;
  ownerName?: string;
  ownerEmail?: string;
}

export interface DocumentStatusResponse {
  id: string;
  status: DocumentStatus;
  processingProgress?: number | null;
  chunkCount?: number | null;
  pageCount?: number | null;
  errorMessage?: string | null;
}